mureader.py

```diff
@@ -0,0 +1,679 @@
+            elif type == MuEnum.ET_MESH_VERTS:
+                for i in range(num_verts):
+                    self.verts.append(mu.read_vector())
+            elif type == MuEnum.ET_MESH_UV:
+                for i in range(num_verts):
+                    self.uvs.append(mu.read_float(2))
+            elif type == MuEnum.ET_MESH_UV2:
+                for i in range(num_verts):
+                    self.uv2s.append(mu.read_float(2))
+            elif type == MuEnum.ET_MESH_NORMALS:
+                for i in range(num_verts):
+                    self.normals.append(mu.read_vector())
+            elif type == MuEnum.ET_MESH_TANGENTS:
+                for i in range(num_verts):
+                    self.tangents.append(mu.read_tangent())
+            elif type == MuEnum.ET_MESH_BONE_WEIGHTS:
+                for i in range(num_verts):
+                    self.boneWeights.append(MuBoneWeight().read(mu))
+            elif type == MuEnum.ET_MESH_BIND_POSES:
+                num_poses = mu.read_int()
+                for i in range(num_poses):
+                    self.bindPoses.append(mu.read_float(16))
+            elif type == MuEnum.ET_MESH_TRIANGLES:
+                num_tris = mu.read_int()
+                tris = []
+                for i in range(int(num_tris / 3)):
+                    tri = mu.read_int(3)
+                    #reverse the triangle winding for Blender (because of the
+                    # LHS/RHS swap)
+                    #avoid putting 0 at the end of the list (Blender doesn't
+                    #like that)
+                    if not tri[0]:
+                        tri = tri[0], tri[2], tri[1]
+                    else:
+                        tri = tri[2], tri[1], tri[0]
+                    tris.append(tri)
+                self.submeshes.append(tris)
+            else:
+                raise ValueError("MuMesh %x %d" % (mu.file.tell(), type))
+        return self
+
+
+class MuRenderer:
+    def __init__(self):
+        self.castShadows = 1
+        self.receiveShadows = 1
+    def read(self, mu):
+        if mu.version > 0:
+            self.castShadows = mu.read_byte()
+            self.receiveShadows = mu.read_byte()
+        num_mat = mu.read_int()
+        self.materials = mu.read_int(num_mat, True)
+        return self
+
+
+class MuSkinnedMeshRenderer:
+    def __init__(self):
+        self.materials = []
+        self.bones = []
+    def read(self, mu):
+        num_mat = mu.read_int()
+        for i in range(num_mat):
+            self.materials.append(mu.read_int())
+        self.center = mu.read_vector()
+        self.size = mu.read_vector()
+        self.quality = mu.read_int()
+        self.updateWhenOffscreen = mu.read_byte()
+        nBones = mu.read_int()
+        for i in range(nBones):
+            self.bones.append(mu.read_string())
+        self.mesh = MuMesh().read(mu)
+        return self
+
+
+class MuCollider_Base:
+    def __init__(self, type):
+        self.type = type
```

Comment:
beware of name shadowing; `type` is a python [builtin](https://docs.python.org/2/library/functions.html#type)